owncloud_service/source_watcher.py

```diff
@@ -0,0 +1,298 @@
+import argparse
+import os
+import time
+from datetime import datetime, timedelta, timezone
+
+import requests
+from requests.auth import HTTPBasicAuth
+from slack_bot import setup_logger
+
+# ----- OwnCloud Configuration -----
```

Comment:
I'm not sure having global variables + argparse + environment variable is wise. 
I'm not a big fan of declaring secrets in the script (because that's prone to commit error, and it requires modifying the script). I would suggest to use only environment variables or an external yaml configuration script to be read at execution (à la SkyPortal).